### ISSUE TITLE
Invalid data received with 433 MHz receiver

### ISSUE THREAD

**Issue by cYnd** [untergasser/tempRainPi#1]
Hi,

I am using the ALecto WS-1200 in the 433 MHz version with a cheap 433 receiver. I am receiving some signals sporadically (it's not like every minute or so but in some random intervals).
Unfortunaley these signals are not correct (or at least they are not correct decoed).

Here are a few examples:

> Signal found:
>11111110011011001111101000111111010110110001101101010100011111110101100111110101010011111001110
 > .   .   .   i   i ?ttttttttttrrrrrrrrRRRRRRRR   .   .   m   m   m   m   d   d   d   d   d   d
>Temp: 91.1°C
> Rain: 10880.7 mm

>Signal found:
>11111110011011001111110100011111111101111011001011100110010011101111010001111001111100011111111
>  .   .   .   i   i ?ttttttttttrrrrrrrrRRRRRRRR   .   .   m   m   m   m   d   d   d   d   d   d
>Temp: 127.9°C
> Rain: 16740.9 mm

> Signal found:
> 11111110011011001111101110101011111110001110110011101110110101001100101111111111011001101111111
>  .   .   .   i   i ?ttttttttttrrrrrrrrRRRRRRRR   .   .   m   m   m   m   d   d   d   d   d   d
> Temp: 109.3°C
> Rain: 9138.0 mm

> Signal found:
> 11111110011011001111110001110100101101111001011101110101001111110011101110110011111011101001111
>   .   .   .   i   i ?ttttttttttrrrrrrrrRRRRRRRR   .   .   m   m   m   m   d   d   d   d   d   d
> Temp: 119.4°C
> Rain: 15617.7 mm

> Signal found:
> 11111110011011001111111110011111001111111110111111100111111111101100111111111110111001111111100
 >  .   .   .   i   i ?ttttttttttrrrrrrrrRRRRRRRR   .   .   m   m   m   m   d   d   d   d   d   d
> Temp: 159.9°C
> Rain: 19017.3 mm

> Signal found:
> 11111110011011001111011100111011011100111101001001010011111111000111111110111110000110100011110
>   .   .   .   i   i ?ttttttttttrrrrrrrrRRRRRRRR   .   .   m   m   m   m   d   d   d   d   d   d
> Temp: 52.5°C
> Rain: 17949.9 mm

> Signal found:
> 11111110011011001111001111010001011010001111011001101010110010011111111111001000111101011101100
>   .   .   .   i   i ?ttttttttttrrrrrrrrRRRRRRRR   .   .   m   m   m   m   d   d   d   d   d   d
> Temp: 8.8°C
> Rain: 9500.4 mm

> Signal found:
> 11111110011011001111110100101111111101000100111001001111011011001011110011110110011110111110111
>  .   .   .   i   i ?ttttttttttrrrrrrrrRRRRRRRR   .   .   m   m   m   m   d   d   d   d   d   d
> Temp: 128.7°C
> Rain: 3070.2 mm

> Signal found:
> 11111110011011001111110001101000100111111101100011111111111100100011011111010010001101101111110
>   .   .   .   i   i ?ttttttttttrrrrrrrrRRRRRRRR   .   .   m   m   m   m   d   d   d   d   d   d
> Temp: 118.8°C
> Rain: 18148.5 mm

> Signal found:
> 11111110011011001111101011011111011001110110111011001111010100111010110111111001111011110101011
>   .   .   .   i   i ?ttttttttttrrrrrrrrRRRRRRRR   .   .   m   m   m   m   d   d   d   d   d   d
> Temp: 99.1°C
> Rain: 14108.1 mm

First I thought maybe the temperature is in fahrenheit but this makes also no sense. Also the rain is wrong.
Are there any known issues? Is it maybe an issue with the frequency and 433 signals need to get decoded differently?

**Comment by EmlJnke** [comment]
For me everything works fine, did You edit the "String to recognize the sensor" to fit for your device? At first I thought I could just remove half of that signal, but the Output was like yours. (just way more frequently 'cause our neighbors seem to have many 433 devices too)

My father has a usb logic Analyzer to visualize 433 MHz signals over a time period, so I used that to record the specific signal of our WS-1200 V2. Although I use very cheap recievers I managed to get a nicely clean signal from that I read the first **19** numbers (19 is important because the 20th is the low battery indicator as we can see in the decoder [here](https://github.com/merbanan/rtl_433/commit/5ed62fc085936024ab324fc8d52692b83ccd871f), so otherwise (like in the script of untergasser) the script would not or only work if the battery is low)

The visualized Signal with the numbers written underneath:
![Bild 18 08 19 um 17 24](https://user-images.githubusercontent.com/42444861/63227732-f4327900-c1e9-11e9-9c46-b8c2a97e6dff.jpeg)

If you have a logic analyzer or some kind of that stuff You can just do the same as i did and then replace the numbers in rscan.pp line 15:
`char idst[] = "11111110011011001110\0";`
with the first 19 numbers of your signal.
so `char idst[] = "<Your_Numbers_Here>\0";`
(You have to re-do this step every time You replace the batteries in the sending thermometer because the ID is newly generated then.)

If you already know all that I'm sorry, but maybe this can help you, good Luck!
Emil

**Comment by cYnd** [comment]
Hey Emil,

thanks for the information. I did not know that I need to adjust the "String to recognize the sensor". Since I don't own a logic analyzer I am wondering if there may be an alternative solution.
Will the rtl_433 project you linked also require these specific numbers recorded by an logic analyzer or will it work "out of the box"?

**Comment by untergasser** [comment]
Hey guys,

incredible that people are using this. It was thought as a test to see if raspi can manage the data handling and to play with replay messages. I thought of merging the read functionality into:
https://github.com/merbanan/rtl_433
Then in a simple setup, raspi could with the simple reciver and rtl_433 decode most 433 sensors.

Unfortunately, I am busy with other projects right now, but I still want to merge both. Else I will write a scanner for this...

Give me two month. 

Best,

Andreas

**Comment by EmlJnke** [comment]
@cYnd 
Yes, as I understood the rtl_433 can just see, by the style of the message, which type of sensor it is. My problem with the Project is that it requires specific USB recievers (so it could possibly work on any normal PC) and is (as far as I know) not planning gpio support in the near future.

@untergasser 
I don't think that merging the Projects would really make sense, because (despite this sensor is already implemented in "rtl_433") your project is perfect and only made for a raspberry pi. As you mentioned it's a proof of concept so it's amazing that it really works that fine (at least for me) and one can use it to add it to other projects or advancing it in a certain way. Decoding and especially understanding such signals are painfully difficult, it's great You make it available here!

(And don't feel forced to advance it now! Of course it would be cool but I thought about how to automatically recognize a sensor's ID myself and as far as I understand c++ it would require _a bit_ of work..)

Cheers

**Comment by cYnd** [comment]
Hi all,

thanks for all the responses - i actually didn't thought anybody will answer :-)

Getting this to work on the raspberry directly would be really awesome, however I already ordered a RXB6 and arduino uno on aliexpress to get the [RFLink](http://www.rflink.nl/blog2/wiring) implementaion working and just connect this via USB to my raspberry and then get the information from there.
